### PR TITLE
DCWL-805-java-11-upgrade

### PR DESCRIPTION
**Pull request by ed1286**

### CODE REVIEW

**Comment by ed1286** [comment]
test this please

**Comment by ed1286** [comment]
test this please

**Comment by ed1286** [comment]
test this please

File: app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala [hunk 1]
```diff
@@ -150,7 +150,7 @@ class CustomsNotificationReceiverController @Inject()(logger : CdsLogger,
 
     val result = statusCode match {
       case code if (code >= 300) && (code < 400) =>
-        Redirect(routes.CustomsNotificationReceiverController.post(), code)
+        Redirect("/customs-notifications-receiver-stub/pushnotifications", code)
```
Comment (IanWood-HMRC) [inline app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala]: is the routes way no longer correct? is hardcoding the way now?

File: build.sbt [hunk 2]
```diff
@@ -31,14 +29,17 @@ lazy val microservice = (project in file("."))
...
+    targetJvm := "jvm-11",
     commonSettings,
     unitTestSettings,
     integrationComponentTestSettings,
     allTest,
     scoverageSettings
   )
   .settings(majorVersion := 0)
-  .settings(scalacOptions += "-P:silencer:pathFilters=routes")
+  .settings(libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Always)
```
Comment (IanWood-HMRC) [inline build.sbt]: libraryDependencySchemes += "org.scala-lang.modules" %% "scala-xml" % VersionScheme.Alway isn't need as already at the bottom

File: app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala (hunk 1 above)
Comment (ed1286) [reply app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala]: for whatever reason it doesn't work anymore. I'm pretty sure the replacement does the same thing. Not exactly sure what changed to break the original

File: build.sbt (hunk 2 above)
Comment (ed1286) [reply build.sbt]: removed 👍

File: app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala (hunk 1 above)
Comment (IanWood-HMRC) [reply app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala]: routes.CustomsNotificationReceiverController.post() works for me.  Perhaps a building issue in your IDE.

File: run_all_tests.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+
+sbt clean scalastyle coverage test IntegrationTest / testOnly coverageReport dependencyUpdates
```
Comment (IanWood-HMRC) [inline run_all_tests.sh]: IntegrationTest isn't found. 

This works though:
sbt clean scalastyle coverage test it:test coverageReport dependencyUpdates

File: build.sbt [hunk 4]
```diff
@@ -31,14 +29,16 @@ lazy val microservice = (project in file("."))
...
+    targetJvm := "jvm-11",
     commonSettings,
     unitTestSettings,
     integrationComponentTestSettings,
     allTest,
     scoverageSettings
   )
   .settings(majorVersion := 0)
-  .settings(scalacOptions += "-P:silencer:pathFilters=routes")
+  .settings(scalacOptions += "-Wconf:cat=unused-imports&src=target/scala-2\\.13/routes/.*:s")
```
Comment (IanWood-HMRC) [inline build.sbt:41]: This is to stop these warnings right?
[warn] /home/ian/work/hmrc/customs-notifications-receiver-stub/conf/routes: Unused import

Comment (ed1286) [reply build.sbt:41]: think so yeah

File: app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala (hunk 1 above)
Comment (ed1286) [reply app/uk/gov/hmrc/customs/notification/receiver/controllers/CustomsNotificationReceiverController.scala]: interesting, reverted 👍

**Review by IanWood-HMRC — APPROVED** [review]

**Pull request merged**